pkg/providers/instancetype/suite_test.go

```diff
@@ -1091,6 +1092,59 @@ var _ = Describe("InstanceType Provider", func() {
 		})
 	})
 
+	Context("Bootstrap", func() {
+		It("should gate kubelet flags that are dependent on kubelet version", func() {
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+			pod := coretest.UnschedulablePod()
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
+			ExpectScheduled(ctx, env.Client, pod)
+
+			Expect(azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Len()).To(Equal(1))
+			vm := azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
+			customData := *vm.Properties.OSProfile.CustomData
+			Expect(customData).ToNot(BeNil())
+			decodedBytes, err := base64.StdEncoding.DecodeString(customData)
+			Expect(err).To(Succeed())
+			decodedString := string(decodedBytes[:])
+			Expect(decodedString).To(ContainSubstring("CREDENTIAL_PROVIDER_DOWNLOAD_URL"))
+			kubeletFlags := decodedString[strings.Index(decodedString, "KUBELET_FLAGS=")+len("KUBELET_FLAGS="):]
+			parseKubeletFlags := func(flags string) map[string]string {
+				flagMap := make(map[string]string)
+				flagList := strings.Split(flags, " --")
+				for _, flag := range flagList {
+					parts := strings.SplitN(flag, "=", 2)
+					if len(parts) == 2 {
+						flagMap["--"+parts[0]] = parts[1]
+					}
+				}
+				return flagMap
+			}
+
+			// TODO: (bsoghigian) leverage the helpers from the azure cni pr once they get in instead for testing kubelet flags
+			flagMap := parseKubeletFlags(kubeletFlags)
+			// NOTE: env.Version may differ from the version we get for the apiserver
+			k8sVersion, err := azureEnv.ImageProvider.KubeServerVersion(ctx)
+			Expect(err).To(BeNil())
+			parsed := semver.MustParse(k8sVersion)
+
+			Expect(decodedString).To(ContainSubstring(
+				fmt.Sprintf("https://acs-mirror.azureedge.net/cloud-provider-azure/%s/binaries/azure-acr-credential-provider-linux-amd64-v%s.tar.gz", parsed.String(), parsed.String()),
+			))
```

Comment:
Should we also gate populating this variable behind 1.30?

Reply:
This would make sense to me, assuming we are not switching previous versions to out-of-tree credential providers. 

Somewhat related, it would be good to have a single helper (something like `usesOutOfTreeCredentialProvider`) instead of sprinkling version checks in different places.